controllers/storagecluster/cephcluster.go

```diff
@@ -528,8 +528,8 @@ func getNetworkSpec(sc ocsv1.StorageCluster) rookCephv1.NetworkSpec {
 	// respect both the old way and the new way for enabling HostNetwork
 	networkSpec.HostNetwork = networkSpec.HostNetwork || sc.Spec.HostNetwork
 
-	// Enable the msgr2 port always if it's not an external cluster
-	if !sc.Spec.ExternalStorage.Enable {
+	// Don't require the msgr2 port if it's an provider or external/consumer cluster
+	if !(sc.Spec.AllowRemoteStorageConsumers || sc.Spec.ExternalStorage.Enable) {
```

Comment:
May I suggest to rewrite the condition in the following way 
```suggestion
	if !sc.Spec.AllowRemoteStorageConsumers && !sc.Spec.ExternalStorage.Enable {
```
To me, this change might convey the condition in a more human-readable way

Reply:
done